integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py

```diff
@@ -381,13 +443,33 @@ def _set_up_collection(
             # Qdrant local raises ValueError if the collection is not found, but
             # with the remote server UnexpectedResponse / RpcError is raised.
             # Until that's unified, we need to catch both.
-            self._recreate_collection(collection_name, distance, embedding_dim, on_disk)
+            self._recreate_collection(collection_name, distance, embedding_dim, on_disk, use_sparse_embeddings)
             # Create Payload index if payload_fields_to_index is provided
             self._create_payload_index(collection_name, payload_fields_to_index)
             return
-
-        current_distance = collection_info.config.params.vectors.distance
-        current_vector_size = collection_info.config.params.vectors.size
+        if self.use_sparse_embeddings:
+            current_distance = collection_info.config.params.vectors[DENSE_VECTORS_NAME].distance
+            current_vector_size = collection_info.config.params.vectors[DENSE_VECTORS_NAME].size
+        if not self.use_sparse_embeddings:
+            current_distance = collection_info.config.params.vectors.distance
+            current_vector_size = collection_info.config.params.vectors.size
+
+        if self.use_sparse_embeddings and not isinstance(collection_info.config.params.vectors, dict):
+            msg = (
+                f"Collection '{collection_name}' already exists in Qdrant, "
+                f"but it has been originaly created without sparse embedding vectors."
+                f"If you want to use that collection, either set `use_sparse_embeddings=False` "
+                f"or run a migration script "
+                f"to use Named Dense Vectors (`text-sparse`) and Named Sparse Vectors (`text-dense`)."
+            )
+            raise ValueError(msg)
+        if not self.use_sparse_embeddings and isinstance(collection_info.config.params.vectors, dict):
+            msg = (
+                f"Collection '{collection_name}' already exists in Qdrant, "
+                f"but it has been originaly created with sparse embedding vectors."
+                f"If you want to use that collection, please set `use_sparse_embeddings=True`"
+            )
+            raise ValueError(msg)
```

Comment:
Explicitly throw an error if there is a possible imcompatibility between legacy vector type and the new vector type (dict) that support sparse embedding